src/Trash/AlbumTrashItemHandler.php

```diff
@@ -0,0 +1,121 @@
+use Sulu\Bundle\TrashBundle\Application\TrashItemHandler\RestoreTrashItemHandlerInterface;
+use Sulu\Bundle\TrashBundle\Application\TrashItemHandler\StoreTrashItemHandlerInterface;
+use Sulu\Bundle\TrashBundle\Domain\Model\TrashItemInterface;
+use Sulu\Bundle\TrashBundle\Domain\Repository\TrashItemRepositoryInterface;
+use Sulu\Component\Security\Authentication\UserInterface;
+
+final class AlbumTrashItemHandler implements
+    StoreTrashItemHandlerInterface,
+    RestoreTrashItemHandlerInterface,
+    RestoreConfigurationProviderInterface
+{
+    private TrashItemRepositoryInterface $trashItemRepository;
+    private EntityManagerInterface $entityManager;
+    private DoctrineRestoreHelperInterface $doctrineRestoreHelper;
+
+    public function __construct(
+        TrashItemRepositoryInterface $trashItemRepository,
+        EntityManagerInterface $entityManager,
+        DoctrineRestoreHelperInterface $doctrineRestoreHelper
+    ) {
+        $this->trashItemRepository = $trashItemRepository;
+        $this->doctrineRestoreHelper = $doctrineRestoreHelper;
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @param Album $album
+     */
+    public function store(object $album, array $options = []): TrashItemInterface
+    {
+        $image = $album->getImage();
+        $creator = $album->getCreator();
+
+        $data = [
+            'title' => $album->getTitle(),
+            'tracklist' => $album->getTracklist(),
+            'imageId' => $image ? $image->getId() : null,
+            'created' => $album->getCreated()->format('c'),
+            'creatorId' => $creator ? $creator->getId() : null,
+        ];
+
+        return $this->trashItemRepository->create(
+            Album::RESOURCE_KEY,
+            (string) $album->getId(),
+            $album->getTitle(),
+            $data,
+            null,
+            $options,
+            Album::SECURITY_CONTEXT,
+            null,
+            null
+        );
+    }
+
+    public function restore(TrashItemInterface $trashItem, array $restoreFormData = []): object
+    {
+        /**
+         * @var array{
+         *     title: string,
+         *     imageId: int|null,
+         *     created: string,
+         *     creatorId: int|null,
+         *     tracklist: array<array{
+         *         type: string,
+         *         title: string|null,
+         *         interpreter: string|null,
+         *     }>,
+         * } $data
+         */
+        $data = $trashItem->getRestoreData();
+
+        $album = new Album();
+        $album->setTitle($data['title']);
+        $album->setTracklist($data['tracklist']);
+        $album->setCreated(new \DateTime($data['created']));
+
+        if ($data['imageId']) {
+            $album->setImage($this->entityManager->find(MediaInterface::class, $data['imageId']));
+        }
+
+        if ($data['creatorId']) {
+            $album->setCreator($this->entityManager->find(UserInterface::class, $data['imageId']));
```

Comment:
```suggestion
            $album->setCreator($this->entityManager->find(UserInterface::class, $data['creatorId']));
```

Reply:
Good catch, thank you! I fixed this in the initial commit.